### PR TITLE
Initial version

### PR DESCRIPTION
**Pull request by Vlaaaaaaad**
Copying files over from [Vlaaaaaaad/tflint-action](https://github.com/Vlaaaaaaad/tflint-action) as per https://github.com/reviewdog/reviewdog/issues/322.

@haya14busa I added an MIT License, is that ok?

Anything else that I should change?

### CODE REVIEW

**Comment by Vlaaaaaaad** [comment]
Tested on another repo and confirmed working so I am marking this as ready for review.

File: action.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+name: 'Run tflint with reviewdog'
+description: '🐶 Run tflint with reviewdog on pull requests to enforce best practices'
+author: 'vlaaaaaaad (reviewdog)'
+
+inputs:
```
Comment (haya14busa) [inline action.yml:5]: Can you add `reporter` to change reviewdog reporter between github-pr-review and github-pr-check for the consistency? 
I added `reporter` flag to actions I created recently (e.g.
https://github.com/reviewdog/action-shellcheck) and I plan to add the same flag to existing actions which don't have the flag yet.

**Review by haya14busa — CHANGES_REQUESTED** [review]
Awesome!

**Comment by haya14busa** [comment]
> @haya14busa I added an MIT License, is that ok?

:+1:

File: action.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  github_token:
+    description: 'GITHUB_TOKEN'
+    required: true
+  args:
+    description: 'List of arguments to send to tflint'
+    default: '--module --format=checkstyle'
```
Comment (haya14busa) [inline action.yml]: If users changed the args and forgot to include `--format`, this action would be broken and I'd suggest that specify `--format=checktyle` in entrypoint.sh and add a warning in the description to warn users that they should not change `--format` flag.

File: action.yml (hunk 1 above)
Comment (haya14busa) [reply action.yml:5]: Actually, I think it's optional for the initial version, You can keep it as is in this PR if you want.

Comment (Vlaaaaaaad) [reply action.yml:5]: I like the idea, adding it.

File: action.yml (hunk 2 above)
Comment (Vlaaaaaaad) [reply action.yml]: Some people might want to use another format too so I don't think I'd help by enforcing it.

For example, I tried to use `--format=json` to get the link which is ignored with `--format=checkstyle` and failed fabulously, but someone else might figure it out.

A warning is definitely helpful so I am adding that.

Comment (Vlaaaaaaad) [reply action.yml]: Nope, that's false. I am enforcing `reviewdog -f=checkstyle` and making that an option seems like too much.

I think leaving it like this is the best option for now.

Comment (haya14busa) [reply action.yml]: If someone else figured it out in the future, we need to bump up major version and also we need to warn users to change the flag to remove or modify --format flag if we leave it as is.

If we put `--format` flag in entrypoint.sh, we can just change the flag along with reviewdog `-f` flag and it won't be breaking changes and we can just bump minor/patch version.

File: action.yml (hunk 1 above)
Comment (Vlaaaaaaad) [reply action.yml:5]: Added, tested.

File: action.yml (hunk 2 above)
Comment (Vlaaaaaaad) [reply action.yml]: Agreed!

Added, tested and confirmed working with the example in `README`.

File: entrypoint.sh [hunk 3]
```diff
@@ -0,0 +1,8 @@
+#!/bin/sh
+
+cd "$GITHUB_WORKSPACE"
+
+export REVIEWDOG_GITHUB_API_TOKEN="${INPUT_GITHUB_TOKEN}"
+
+tflint --format=checkstyle ${INPUT_ARGS} \
```
Comment (haya14busa) [inline entrypoint.sh]: INPUT_FLAGS

**Review by haya14busa — CHANGES_REQUESTED** [review]

**Comment by Vlaaaaaaad** [comment]
😞 Apologies about the flurry of commits -- my eyes fail me.

File: entrypoint.sh (hunk 3 above)
Comment (Vlaaaaaaad) [reply entrypoint.sh]: 🤦‍♂ fixed.

File: README.md [hunk 4]
```diff
@@ -1,2 +1,82 @@
...
+
+Optional. Report level for reviewdog [`info`,`warning`,`error`].
+It's same as `-level` flag of reviewdog.
+The default is `error`.
+
+### `reporter`
+
+Optional. Reporter of reviewdog command [`github-pr-check`,`github-pr-review`].
+The default is `github-pr-check`.
+`github-pr-review` can use Markdown and add a link to rule page in reviewdog reports.
```
Comment (haya14busa) [inline README.md]: [nit] It's still not true, so can you remove this sentence?

**Review by haya14busa — APPROVED** [review]
LGTM :+1: ✨

File: README.md (hunk 4 above)
Comment (Vlaaaaaaad) [reply README.md]: Done!

**Review by haya14busa — APPROVED** [review]
LGTM++

Feel free to merge it by yourself (or should I?).

I personally haven't used squash & merge in GitHub and maybe it's better for you to write merge commit message?

**Comment by Vlaaaaaaad** [comment]
I always used squash & merge 😆 Merged! Thank you so much!

I'll create a release now to have it show in the Marketplace -- never done this so we'll see how it goes.

**Comment by haya14busa** [comment]
Thank you for your work! 🚀 

> I'll create a release now to have it show in the Marketplace -- never done this so we'll see how it goes.

SG. I'll go to bed so I cannot reply for a while but please ask me anything if you got troubled.

**Pull request merged**